src/components/Spawn/spawnHostModal/useVirtualWorkstationDefaultExpiration.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from "react";
+import { usePrevious } from "hooks";
+import { getDefaultExpiration } from "./getFormSchema";
+
+export const useVirtualWorkstationDefaultExpiration = ({
```

Comment:
During a migration, we should permit users to create a new expirable host even if they already have reached `unexpirableHostsPerUser`, since the current host will be set to expire shortly. I think this hook may be a good place to handle that!

Reply:
hmm. I would be cautious about this assumption/feature. This seems like an easy way to run into a scenario where a user can spawn unlimited virtual workstations by just creating new volumes and then updating the expiration time on the old hosts. Unless I am missing something?

Reply:
Good point -- I don't think we need to assume super malicious user intent (especially since the back end doesn't even prevent this) but we can make it a little tougher to spawn infinite hosts. We could permit this only on the migration modal if the user has unexpiredHostCount <= unexpirableHostsPerUser (as opposed to just <), thereby allowing them to create 1 more unexpirable host but not opening it up entirely. cc @SupaJoon

Reply:
Ah, and we prevent editing a host to be unexpirable if 2 already exist :)
<img width="873" alt="image" src="https://user-images.githubusercontent.com/9298431/199547599-973c2832-56dd-42dd-bc23-e8fd3c75084d.png">